wadm/lib/wadm/deployments/deployment_monitor.ex

```diff
@@ -43,24 +59,110 @@ defmodule Wadm.Deployments.DeploymentMonitor do
 
+  @impl true
+  def handle_call(:get_recon_status, _from, state) do
+    {:reply, state.recon_state, state}
+  end
+
   @impl true
   def handle_continue(:ensure_lattice_supervisor, state) do
     # Make sure that there's a lattice supervisor running
     {:ok, _pid} = Wadm.LatticeSupervisor.start_lattice_supervisor(state.lattice_id)
 
+    Process.send_after(self(), :reconcile_initializing, 45_000)
+
     {:noreply, state}
   end
 
+  @impl true
+  def handle_info(:remove_backoff, state) do
+    {:noreply, %State{state | recon_state: :idle}}
+  end
+
+  @impl true
+  def handle_info(:reconcile_initializing, state) do
+    # TODO - deal with error state
+    if state.recon_state == :initializing do
+      pid = Wadm.LatticeStateMonitor.get_process(state.lattice_id)
+
+      if pid != nil do
+        {cmd_count, _error_count} = do_reconcile(state.spec, state.lattice_id)
+
+        if cmd_count > 0 do
+          {:noreply, %State{state | recon_state: :reconciling}}
+        else
+          {:noreply, %State{state | recon_state: :idle}}
+        end
+      else
+        {:noreply, %State{state | recon_state: :idle}}
+      end
+    else
+      {:noreply, state}
+    end
+  end
+
+  @impl true
+  def handle_info({:lattice_changed, lattice, _event}, state) do
+    # in backoff/cooldown mode, incoming events have no effect
+    if state.recon_state == :reconciling do
+      {:noreply, state}
+    else
+      Logger.debug(
+        "Deployment monitor #{state.spec.name} handling state change for lattice #{lattice.id}"
+      )
+
+      # TODO make use of the error state
+      _ = do_reconcile(state.spec, lattice)
+
+      {:noreply, %State{state | recon_state: :reconciling}}
+    end
+  end
+
+  defp do_reconcile(spec, lattice) do
+    {skips, cmds} =
+      Wadm.Reconciler.AppSpec.reconcile(spec, lattice)
+      |> Enum.split_with(fn command -> command.cmd == :no_action || command.cmd == :error end)
+
+    errors = skips |> Enum.filter(fn command -> command.cmd == :error end)
+
+    if length(errors) > 0 do
+      # TODO - once we move past the naive reconciliation strategy, we might want a
+      # discrete error/failed state
+      Logger.error("Failed to perform reconciliation pass: #{errors |> Enum.join(",")}")
+    end
+
+    Logger.debug("Reconciliation found #{length(skips)} no-ops and #{length(cmds)} commands")
+
+    cmds
+    |> Enum.map(fn cmd ->
+      {Wadm.Reconciler.Command.to_lattice_control_command(spec, lattice.id, cmd), lattice.id}
+    end)
+    |> Enum.each(&publish_lattice_control_command/1)
+
+    # even if we had a failure in the command list, back off so that we don't attempt
+    # to re-reconcile for another 30 seconds
+    Process.send_after(self(), :remove_backoff, 45_000)
+
+    {length(cmds), length(errors)}
+  end
+
+  defp publish_lattice_control_command({{topic, cmd}, lattice_id}) do
+    _ = Gnat.request(String.to_atom(lattice_id), topic, cmd)
+  end
```

Comment:
This would be a good place to error log if a lattice control request fails to receive a reply, or even when a request is malformed